src/features/read/readSlice.ts

```diff
@@ -178,9 +179,51 @@ const selectQuestion = (state: RootState): IQuestion | null => {
 
 export const selectMode = (state: RootState): Mode | null => {
   const treatment = selectTreatment(state);
+  const book = selectBook(state)
+  const bookType = book?.fields.type;
+
+  //check if pre- or post-test
+  if( bookType === "pre-test" || bookType === "post-test" ){
+  //make sure book isn't null (makes getPage feel better)
+    if(book){
+      //if sys.id is a question, then continue. Otherwise, it's narrative page so skip over
+      if(Book.getPage(book,selectPageNumber(state))?.sys.contentType.sys.id === "question"){
```

Comment:
I think there is already a way to do this check that will be a little cleaner. 

Check out `models/BookPage.ts`. I would do:

```ts
const bookPage = Book.getPage(book, selectPageNumber(state))
const determine = BookPage.determineType(bookPage)
if (determine.type === 'page') {
  return null
}
const question = determine.question // You won't need to use `selectQuestion` if you go this route because you will aready have it from the determineType call
```